src/UnitTests/Analysis/ValuePropagationTests.cs

```diff
@@ -471,7 +472,43 @@ public void VpPhiWithConstants()
             Assert.AreEqual("r3 = 0x4711", instr.ToString());
         }
 
-		private class DpbMock : ProcedureBuilder
+        [Test(Description = "(x = phi(<exp>, <exp>, x, x)) ==> (x = <exp>)")]
+        public void VpPhiLoops()
```

Comment:
`all the a_n, after value propagation, are equal`
Not all, just part of them. And the other identifiers could be simplified to `x`